helpers/isChatToYourself.js

```diff
@@ -0,0 +1,28 @@
+const {User} = require('../databases/models');
+const UsersFunction = require('../databases/functions/users');
+
+module.exports = {
+  isChatToYourself: async (my_user_id, member) => {
+    let user = await UsersFunction.findUserById(User, member);
+    if (!user) {
+      return {
+        success: false,
+        message: 'User not found'
+      };
+    }
+    const user_ids = new Set([user.user_id]);
+    if (!user.is_anonymous) {
+      const anon_user = await UsersFunction.findAnonymousUserId(User, user.user_id);
+      user_ids.add(anon_user.user_id);
+    } else {
+      const sign_user = await UsersFunction.findSignedUserId(User, user.user_id);
+      user_ids.add(sign_user.user_id);
+    }
+
+    const isChattingToSelf = user_ids.has(my_user_id);
+    return {
+      success: !isChattingToSelf,
+      message: isChattingToSelf ? 'Cannot chat to yourself' : 'Success'
+    };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The response object structure is inconsistent. In case of an error (like user not found), the `success` field is set to `false` and a `message` is returned. However, when checking if a user is chatting with themselves, the `success` field is set to `!isChattingToSelf`, which can be confusing. It would be better to always return `success: true` when the function executes without errors, and include an additional field, like `isChattingToSelf`, to indicate whether the user is trying to chat with themselves.

```diff
-    const isChattingToSelf = user_ids.has(my_user_id);
-    return {
-      success: !isChattingToSelf,
-      message: isChattingToSelf ? 'Cannot chat to yourself' : 'Success'
-    };

+    return {
+      success: true,
+      isChattingToSelf,
+      message: isChattingToSelf ? 'Cannot chat to yourself' : 'Success'
+    };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->